tests/test_web/test_healthcheck.py

```diff
@@ -0,0 +1,240 @@
+    }
+
+    mock_database_connection_check.assert_called_once()
+    mock_rabbitmq_connection_check.assert_called_once_with(config.broker_url)
+    mock_workers_status.assert_called_once_with(retries=2)
+
+
+@pytest.mark.parametrize('short', [True, False])
+@mock.patch('iib.web.healthcheck.get_worker_config')
+@mock.patch('iib.web.healthcheck.database_connection_check')
+@mock.patch('iib.web.healthcheck.rabbitmq_connection_check')
+@mock.patch('iib.web.healthcheck.workers_status')
+def test_status_database_not_okay(
+    mock_workers_status,
+    mock_rabbitmq_connection_check,
+    mock_database_connection_check,
+    mock_get_worker_config,
+    short,
+):
+
+    config = mock_worker_config()
+
+    mock_get_worker_config.return_value = config
+
+    mock_database_connection_check.return_value = (False, 'database connection failed')
+    mock_rabbitmq_connection_check.return_value = (True, None)
+    mock_workers_status.return_value = [{'name': 'celery@a868abda4f69', 'available': True}]
+
+    if short:
+        with pytest.raises(IIBError, match='DATABASE unavailable: database connection failed'):
+            healthcheck.status(short=True)
+        return
+
+    result = healthcheck.status(short=False)
+
+    expected_services = [
+        {'name': 'DATABASE', 'available': False, 'reason': 'database connection failed'},
+        {'name': 'RABBITMQ', 'available': True},
+    ]
+
+    assert result == {
+        'services': expected_services,
+        'workers': mock_workers_status.return_value,
+    }
+
+    mock_database_connection_check.assert_called_once()
+    mock_rabbitmq_connection_check.assert_called_once_with(config.broker_url)
+    mock_workers_status.assert_called_once_with(retries=2)
+
+
+@pytest.mark.parametrize('short', [True, False])
+@pytest.mark.parametrize(
+    'workers_status, expected_result',
+    [
+        ([], False),
+        ([{'name': 'celery@a868abda4f68', 'available': False, 'reason': 'unknown error'}], False),
+        (
+            [
+                {'name': 'celery@a868abda4f68', 'available': False, 'reason': 'unknown error'},
+                {'name': 'celery@a868abda4f69', 'available': True},
+            ],
+            True,
+        ),
+    ],
+)
+@mock.patch('iib.web.healthcheck.get_worker_config')
+@mock.patch('iib.web.healthcheck.database_connection_check')
+@mock.patch('iib.web.healthcheck.rabbitmq_connection_check')
+@mock.patch('iib.web.healthcheck.workers_status')
+def test_status_workers_not_okay(
+    mock_workers_status,
+    mock_rabbitmq_connection_check,
+    mock_database_connection_check,
+    mock_get_worker_config,
+    short,
+    workers_status,
+    expected_result,
+):
+
+    config = mock_worker_config()
+
+    mock_get_worker_config.return_value = config
+
+    mock_database_connection_check.return_value = (True, None)
+    mock_rabbitmq_connection_check.return_value = (True, None)
+    mock_workers_status.return_value = workers_status
+
+    if short and not expected_result:
```

Comment:
Could add to parametrize a variable error and error_message instead of adding `if` statement?

Reply:
@zanssa This comment was resolved without a response. Could you please explain why you prefer to keep it this way?

Ideally, it would be nice if you could split this into two different tests instead of one. But I am willing to look the other way